snapcraft/storeapi/__init__.py

```diff
@@ -395,11 +395,12 @@ def _update_status(self, queue):
 
     def _get_status(self):
         connection_errors_allowed = 10
-        while connection_errors_allowed:
+        while True:
             try:
                 content = requests.get(self.__status_details_url).json()
             except (requests.ConnectionError, requests.HTTPError) as e:
+                if not connection_errors_allowed:
+                    yield e
```

Comment:
Thanks, if I intend to use a variable or method in a conditional I try to make it readable